driver-core/src/test/functional/com/mongodb/client/model/mql/TypeMqlValuesFunctionalTest.java

```diff
@@ -225,6 +226,35 @@ public void dateAsStringTest() {
                 of(instant).asString(of("America/New_York"), of("%Y-%m-%dT%H:%M:%S.%L%z")));
     }
 
+    /**
+     * Tests that with server default format and non-UTC timezone, {@code $dateToString} won't output invalid ISO 8601 string ending with 'Z'.
+     * Ticket: JAVA-5044
+     */
+    @Test
+    void dateAsStringWithDefaultFormat() {
```

Comment:
This is testing a new API. But, since this is a bugfix, I would expect it to test an existing API, where the test would fail before the prod change, and pass afterward. Is that not the case?